---
title: "2_Exploring_EST_Denovo_Data"
author: "Anna A. Hippmann"
date: "February 3, 2017"
output: 
  html_document: 
    keep_md: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### Exploring EST de novo data

What I would like to do here is explore the denovo data a bit and getting some feel of the subversiveness of redundancies and how we might need to deal with them. The information that we do have about each transcript is if it:

* hits a predicted reference gene on the genome of _T. oceanica_ (CCMP 1005) ( __gene__)

* hits the chloroplast genome ( __chl__)

* hits any non-coding part of the genome ( __genome__)

* hits nothing on the genome ( __not__)

What I would like to explore in more detail

1) What is the proportion of each of the above defined 4 pools (i.e. ORF, chl, gene, not)

2) how much redundencies are there (i.e. in ORF and chl pool, how many transcripts are hitting the same reading frame?) 
  - Those transcripts that are hitting the same genes will be dubbed to belong the same __pool__ of transcripts
  
3) are the redundant transcripts of one pool expressed in a similar way or are there __sub-pools__ of expression

####Why do I want to know/deal with the redundancies?

With my proteomics data I would like to investigate the proteomic response of _T. oceanica_ (CCMP 1003) to low Cu. At the moment I am particularly looking into the carbon metabolism. Earlier work, especially by Allen et al, Gruber et al and Smith et al, has shown that many metabolic pathways include several isozymes that are often times targeted to different cellular subcompartments. Furthermore some are used within the whole cellular metabolism in differnet ways than we would find in plants or animals. All of this makes it crucial to know where a  protein is really targeted to.

_T. oceanica's_ publicly available genome is helpful. However, it is in a rather rudimentary _Scaffold_ state when compared to the golden standard of genomes such as _T. pseudonanas_ or _P. tricornutums_. Oftentimes, gene models in _T. oceanica_ are missing their target sequence. THe EST library could help enhance the current gene models, especially upstream of the genes of interest to me to allow a beeter cellular targeting prediction.


##Libraries used
``` {r libraries, eval=TRUE}
suppressPackageStartupMessages(library(visreg))
suppressPackageStartupMessages(library(dplyr))
library(tidyr)
library(ggplot2)
library(knitr)
#library (gridExtra)
#library(cowplot)
```


```{r data used}

EST_all <- read.delim("Input_Data/T.oceanica_annotation_denovo_2015.txt")

EST_small <- EST_all %>% 
  select(1:9) %>% 
  rename(transcript = orf_id, ref_gene=ref_gene_id, ref_scaffold = ref_genome_id)


names(EST_small)
      
```
####Now I would like to have a summary of each column

```{r summary of columns}
#counts the number of disinct values in each variable
(summary_EST <- EST_small %>% 
  summarise_each(funs(n_distinct)))


summary_gene <- EST_small %>% 
  group_by(ref_gene) %>% 
  summarise(n()) %>% 
  rename(count = `n()` ) %>% 
  arrange(desc(count))

(head(summary_gene))


```

Lets have a closer look at [THAOC_35840](https://www.ncbi.nlm.nih.gov/protein/397567357).

As per NCBI, this gene is putatively a "sterol regulatory element-binding protein, partial". It is hit by 757 transcripts
```{r THAOC_35840-pool}
THAOC_35840 <- EST_small %>% 
  filter(ref_gene == "THAOC_35840")
```

So, the transcript_pool for THAOC_35840 consiste of `length(unique(THAOC_35840$transcript ))` trancripts.

As mentioned above, the state of the officially available TO genome is still in the scaffold phase. And the scaffolds harbouring usually 1-2 predicted proteins only. When we look at our data, we see in the summary, that the transcripts hit `summary_EST[1,4]` ref_genes and `summary_EST[1,5]` reference scaffolds.

When we look our THAOC_35840 transcript pool, they all hit the same reference gene, but when we look at the scaffolds, they hit combined `length(unique(THAOC_35840$ref_scaffold ))`. How can this be?